src/web/app/components/course-edit-form/course-edit-form.component.spec.ts

```diff
@@ -1,35 +1,177 @@
+import { CopyCourseModalComponent } from '../copy-course-modal/copy-course-modal.component';
+import { LoadingRetryModule } from '../loading-retry/loading-retry.module';
+import { LoadingSpinnerModule } from '../loading-spinner/loading-spinner.module';
+import {
+  CourseAddFormModel,
+   CourseEditFormMode,
+   DEFAULT_COURSE_ADD_FORM_MODEL,
+} from './course-edit-form-model';
 import { CourseEditFormComponent } from './course-edit-form.component';
 
 describe('CourseEditFormComponent', () => {
   let component: CourseEditFormComponent;
   let fixture: ComponentFixture<CourseEditFormComponent>;
+  let feedbackSessionsService: FeedbackSessionsService;
+  let ngbModal: NgbModal;
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      declarations: [
-        CourseEditFormComponent,
-        AjaxLoadingComponent,
-      ],
+  const validCourseId: string = 'CS1101S';
+  const validTimeZone: string = 'Asia/Singapore';
+  const validInstitute: string = 'Test Institute';
+  const testTimeZone: string = 'Australia/Adelaide';
+  const timeZoneOffsets1: Record<string, number> = { GMT: 570 };
+  const testCourse: Course = {
+    courseId: 'testId',
+    courseName: 'Test Course',
+    timeZone: validTimeZone,
+    institute: validInstitute,
+    creationTimestamp: 0,
+    deletionTimestamp: 1000,
+  };
+
+  const spyStatusMessageService: any = {
+    showErrorToast: jest.fn(),
+    showSuccessToast: jest.fn(),
+  };
+  const timezoneServiceStub: any = {
+    getTzOffsets: jest.fn(() => timeZoneOffsets1),
+    guessTimezone: jest.fn(() => testTimeZone),
+  };
+
+  const spyCourseService: any = {
+    createCourse: jest.fn(() => of({})),
+    getAllCoursesAsInstructor: jest.fn(() => of({ courses: [testCourse] })),
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [CourseEditFormComponent],
       imports: [
         HttpClientTestingModule,
         FormsModule,
+        ReactiveFormsModule,
+        RouterTestingModule,
+        NgbModule,
+        AjaxLoadingModule,
+        CommonModule,
+        FormsModule,
+        AjaxLoadingModule,
+        LoadingRetryModule,
+        LoadingSpinnerModule,
+      ],
+      providers: [
+        { provide: StatusMessageService, useValue: spyStatusMessageService },
+        { provide: CourseService, useValue: spyCourseService },
+        { provide: TimezoneService, useValue: timezoneServiceStub },
       ],
     })
-      .compileComponents();
-  });
+    .compileComponents();
+  }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(CourseEditFormComponent);
+    feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
+    ngbModal = TestBed.inject(NgbModal);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
+    fixture.detectChanges();
     expect(component).toBeTruthy();
   });
+
+  it('should snap when in ADD Mode and is display only', () => {
```

Comment:
Maybe we can add some tests for edit mode as well?

Reply:
Will do!